src/main/java/com/google/ehub/servlets/ProfileServlet.java

```diff
@@ -17,46 +21,58 @@
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
 
-/** 
-* Handles adding the form submission to the datastore in 
-* POST and returns json of the user's profile information
-* to client.
-*/
+/**
+ * Handles adding the form submission to the datastore in
+ * POST and returns json of the user's profile information
+ * to client.
+ */
 @WebServlet("/profile-data")
 public class ProfileServlet extends HttpServlet {
-  
   private static final String NAME_PROPERTY_KEY = "name";
   private static final String EMAIL_PROPERTY_KEY = "email";
   private static final String USERNAME_PROPERTY_KEY = "username";
   private static final String BIO_PROPERTY_KEY = "bio";
-  ProfileDatastore profileData = new ProfileDatastore();
+  private static final String EDIT_PROPERTY_KEY = "edit";
+  
+  private static final UserService userService = UserServiceFactory.getUserService();
+  private static final ProfileDatastore profileData = new ProfileDatastore();
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    
-
+    String email = userService.getCurrentUser().getEmail();
     String name = request.getParameter(NAME_PROPERTY_KEY);
-    String email = request.getParameter(EMAIL_PROPERTY_KEY);
-    String username = request.getParameter(USERNAME_PROPERTY_KEY );
+    String username = request.getParameter(USERNAME_PROPERTY_KEY);
     String bio = request.getParameter(BIO_PROPERTY_KEY);
+    boolean edit = Boolean.parseBoolean(request.getParameter(EDIT_PROPERTY_KEY));
 
-    if (!areValidParameters(name, email, username, bio)) {
-      System.err.println("ProfileServlet: Post Request parameters empty");
-      return;
+    if (edit) {
+      profileData.editProfile(name, username, bio);
+    } else {
+      if (!areValidParameters(name, username, bio)) {
+        response.sendError(HttpServletResponse.SC_BAD_GATEWAY, "Post Request parameters empty");
+        return;
+      }
+      profileData.addUserProfileToDatastore(name, email, username, bio);
     }
 
-    profileData.addUserProfileToDatastore(name, email, username, bio);
     response.sendRedirect("/ProfilePage.html");
   }
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    LoginServlet login = new LoginServlet();
-    String useremail = login.getEmail();
-    UserProfile newUser = profileData.getUserProfile(useremail);
+    if (!userService.isUserLoggedIn()) {
+      response.sendError(HttpServletResponse.SC_BAD_GATEWAY, "User must logged in");
+    } else {
+      String userEmail = userService.getCurrentUser().getEmail();
+      UserProfile newUser = profileData.getUserProfile(userEmail);
```

Comment:
super tiny nit: does `newUser` make sense here? Maybe rename to `userProfile` which is a little more accurate since it doesn't represent a new user per se.